pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/OfflineSegmentAssignment.java

```diff
@@ -52,6 +45,12 @@ protected int getReplication(TableConfig tableConfig) {
   @Override
   public List<String> assignSegment(String segmentName, Map<String, Map<String, String>> currentAssignment,
       Map<InstancePartitionsType, InstancePartitions> instancePartitionsMap) {
+    // Check for Dim table
```

Comment:
I checked back this, For the dim table, the problem is the assignment flow goes from OfflineSegmentAssigment before actual segment assignment strategy is set in BaseAssignment. So pre conditions fail because instance partitions is null. So we would have to check for Dim table in OfflineSegmentAssignment. wdyt?